src/renderer/hooks/use-app-settings.ts

```
@@ -1,9 +1,77 @@
+const panelWriteRequestIds: Record<PanelSettingKey, number> = {
+  sidebarVisible: 0,
+  fileExplorerVisible: 0
+}
+let pendingPanelWriteCount = 0
+let pendingPanelWriteWaiters: Array<() => void> = []
+
+function notifyPanelWriteSettled(): void {
+  if (pendingPanelWriteCount === 0 && pendingPanelWriteWaiters.length > 0) {
+    const waiters = pendingPanelWriteWaiters
+    pendingPanelWriteWaiters = []
+    waiters.forEach((resolve) => resolve())
```

Biome (2.4.4)

[error] This callback passed to forEach() iterable method should not return a value.

(lint/suspicious/useIterableCallbackReturn)